fix(ui): update plugin documentation tag design to match no-code interface

Pratapchandradeo opened the pull request:
## What does this PR do?
Updates the plugin documentation tag design to align with the no-code interface styling as requested in issue #12178.

## Changes made:
- Applied no-code `.type-tag` styling to plugin documentation `.type-box` elements
- Updated property name typography to use monospace font for consistency
- Improved required indicator (*) visibility
- Ensured consistent spacing and design tokens
- Removed conflicting CSS rules that were overriding the tag styles

## Files modified:
- `ui/src/styles/components/plugin-doc.scss`

## Before:
<img width="1744" height="918" alt="image" src="https://github.com/user-attachments/assets/171f8876-0af9-4de0-9310-d50dbe88329f" />

## After:
<img width="1576" height="915" alt="image" src="https://github.com/user-attachments/assets/277c6d0b-755d-40d3-90e6-47fb18d60548" />


Fixes #12178

Pratapchandradeo committed:
fix(ui): update plugin doc tag design to match no-code interfacegit commit -m
- Applied no-code .type-tag styling to plugin documentation type boxes
- Updated property name typography to use monospace font
- Improved required indicator visibility
- Ensured consistent spacing and design tokens
Fixes #12178

Piyush-r-bhaskar commented on ui/src/styles/components/plugin-doc.scss:
```diff
@@ -65,10 +66,6 @@
                 color: var(--ks-content-link);
             }
 
-            [id$="-body"]:not(#examples-body) span {
```
restore it please

Pratapchandradeo replied:
Hi @Piyush-r-bhaskar , thanks for the review! 

I understand you want me to restore the `[id$="-body"]:not(#examples-body) span` rule. However, this rule sets `font-size: 1rem !important` which overrides the 12px styling needed for the tag design to match the no-code interface.

**The issue:** 
- No-code tags use `font-size: 12px`
- The restored rule would force `font-size: 1rem` (usually 16px)
- This would break the tag design consistency

**Could we either:**
1. Keep the rule, but exclude `.type-box` elements from it?
2. Or modify the rule to not affect tag elements?

What would you prefer? I'm happy to implement either solution! 🙏

Piyush-r-bhaskar replied:
I would prefer to change the style or size from the `ui-libs` repository where the type-box is actually implemented. 

But since it's style changes you should be able to change it from this repo as well.

Pratapchandradeo committed:
fix: update plugin doc tag design with no-code styling
- Applied no-code tag design with proper 12px font size
- Maintained restored [id$='-body'] span rule as requested
- Enhanced both simple tags and reference tags
- Fixed alignment and spacing for consistent design
Fixes #12178

Pratapchandradeo committed:
Merge branch 'develop' into fix/plugin-doc-tag-design

Pratapchandradeo commented:
Hi @Piyush-r-bhaskar ,

I understand your preference for fixing this in the ui-libs repository. However, I encountered several challenges when trying to modify ui-libs directly:

## Challenges with ui-libs Approach:

1. **Compiled Dependencies**: The ui-libs in `node_modules` is a compiled, read-only version - changes don't reflect during development
2. **Development Workflow**: Without access to the source ui-libs repository, I can't test or verify changes in the actual component
3. **Immediate Solution Needed**: This approach provides a working solution that can be tested and verified now

## Current Implementation Benefits:

✅ **Immediate Fix**: Works in the current development environment  
✅ **Preserves Intent**: Maintains your restored CSS rule while achieving design goals  
✅ **Testable**: Can be verified and reviewed immediately  
✅ **Reversible**: Easy to migrate to ui-libs when the repository is accessible

## The CSS override approach:

- Uses proper specificity to target only the tag elements
- Maintains the `font-size: 1rem` rule for other span elements
- Matches the no-code interface design exactly

This provides a practical solution that works within our current constraints while achieving the design alignment requested in #12178.

Could you please review the current implementation? If the ui-libs repository becomes accessible, I'd be happy to contribute the proper fix there as well.

## Before:

<img width="1359" height="710" alt="image" src="https://github.com/user-attachments/assets/13787b99-9fbf-47bd-9497-25aaefd48a6b" />

## After:

<img width="1435" height="702" alt="image" src="https://github.com/user-attachments/assets/3b8d0b56-ceb1-4ef2-86fc-c5985a5a8bfd" />



Ready for your review! 🙏

MilosPaunovic commented:
@Pratapchandradeo Can you please not use AI to write this long descriptions and messages, just write with your own words what was done.

Pratapchandradeo commented:
Hi @MilosPaunovic  ,
Thanks for the feedback. I’ll keep the descriptions shorter and more direct next time.

 This PR fixes the tag design issue by applying a scoped CSS override that keeps the intended font size and matches the no-code UI design. Once ui-libs access is available, I’ll move the fix there properly.

MilosPaunovic commented:
What do you mean by `ui-libs` access be available?

Pratapchandradeo commented:
The ui-libs is in node_modules as a compiled package - it's read-only and changes don't reflect during development. That's why I used a CSS override for now.

The proper fix should be in the ui-libs source code, but this CSS solution works to fix the tag design immediately.

MilosPaunovic commented:
So why not handle it in https://github.com/kestra-io/ui-libs?

Pratapchandradeo commented:
Thanks for the link. Since this was my 2nd contribution, I started fixing it here because the issue appeared in this repo, and I didn’t have direct access or setup for ui-libs earlier.

I’ll check out the ui-libs repository you shared and try to move the fix there instead.

Piyush-r-bhaskar commented:
I just see we probably need the style tweaks only for tag without eye icon. Awaiting confirmation from our designer https://github.com/kestra-io/kestra/issues/12178#issuecomment-3435412909

and Yes as Milos said, I don't see any challenges for it to be done properly in `ui-libs` repo and that's ideal.

Pratapchandradeo commented:
Thanks @Piyush-r-bhaskar  and @MilosPaunovic
Got it. I’ll wait for the designer’s confirmation on whether the style tweaks are needed only for tags without the eye icon.

Once that’s confirmed, I’ll move the changes to the ui-libs repo as suggested.
Appreciate both of your guidance!

Piyush-r-bhaskar commented:
@Pratapchandradeo  here is the feedback, you just need to make changes to normal tag not the one with eye icon -  https://github.com/kestra-io/kestra/issues/12178#issuecomment-3435426381

and the change will be only in `ui-libs` repository. So, closing this PR and will wait for your PR in `ui-libs` repo. Thanks for your work.
